test/test_learn.py

```diff
@@ -1,260 +1,78 @@
-                'Database %s not found. Make sure to run test_generate before')
+def test_learn():
+    """ This test will simply run deeprank's learning code. It doesn't
+        test any particular feature or target classes.
 
-        # clean the output dir
-        out = './out_3d_fly'
-        if os.path.isdir(out):
-            for f in glob.glob(out + '/*'):
-                os.remove(f)
-            os.removedirs(out)
+        The result of deeprank's learning is not verified. This test
+        only runs the code to be sure there are no exceptions thrown.
+    """
 
-        # declare the dataset instance
-        data_set = DataSet(
-            database,
-            test_database=None,
-            chain1='C',
-            chain2='D',
-            mapfly=True,
-            use_rotation=2,
-            grid_info={
-                'number_of_points': (10, 10, 10),
-                'resolution': (3, 3, 3)},
-            select_feature={
-                'AtomicDensities': {'C': 1.7, 'N': 1.55, 'O': 1.52, 'S': 1.8},
-                'Features': ['coulomb', 'vdwaals', 'charge', 'PSSM_*']},
-            select_target='DOCKQ',
-            tqdm=True,
-            normalize_features=False,
-            normalize_targets=False,
-            clip_features=False,
-            pair_chain_feature=np.add,
-            dict_filter={'DOCKQ': '<1'})
-        # dict_filter={'IRMSD':'<4. or >10.'})
+    feature_modules = ["test.feature.feature1", "test.feature.feature2"]
+    target_modules = ["test.target.target1"]
+    pdb_path = "test/101M.pdb"
+    pssm_paths = {"A": "101M.A.pdb.pssm"}
 
-        # create the networkt
-        model = NeuralNet(data_set, cnn3d, model_type='3d', task='reg',
-                          cuda=False, plot=True, outdir=out)
+    atomic_densities = {'C': 1.7, 'N': 1.55, 'O': 1.52, 'S': 1.8}
+    grid_info = {
+       'number_of_points': [30,30,30],
+       'resolution': [1.,1.,1.],
+       'atomic_densities': atomic_densities,
+    }
 
-        # start the training
-        model.train(
-            nepoch=5,
-            divide_trainset=0.8,
-            train_batch_size=2,
-            preshuffle_seed=2019,
-            num_workers=0)
+    mutant = PdbMutantSelection(pdb_path, "A", 10, "C", pssm_paths)
 
-    @unittest.skipIf(skip, "torch fails on Travis")
-    @staticmethod
-    def test_learn_3d_reg():
-        """Use a 3D CNN for regularization."""
+    work_dir_path = mkdtemp()
+    try:
+        hdf5_path = os.path.join(work_dir_path, "test.hdf5")
 
-        # adress of the database
-        train_database = '1ak4.hdf5'
-        if not os.path.isfile(train_database):
-            raise FileNotFoundError(
-                'Database %s not found. Make sure to run test_generate before',
-                train_database)
+        # data_augmentation has been set to a high number, so that
+        # the train, valid and test set can be large enough.
+        data_generator = DataGenerator([mutant], data_augmentation=50,
```

Comment:
How is the number for data augmentation arrived at?

Reply:
I needed a set of input data large enough to train the network on. So that there's room for a train, validation and test set. At the same time, the larger the set, the longer the test takes. So I decided on 50. It could be smaller I guess.

Reply:
Ok, sure. I think we would have some idea on this once we have data from BioProdict.